components/SchemaGenerator/SchemaGenerator.tsx

```diff
@@ -0,0 +1,77 @@
+import Head from 'next/head';
+import type { JobPost } from 'lib/peopleHR';
+
+export const SchemaGenerator = ({ job }: { job: JobPost }) => {
+    let splittingString = ' - ';
+    if (job.salaryRange.includes(' to ')) {
+        splittingString = ' to ';
+    }
+
+    const minSalary = job.salaryRange.split(splittingString)[0];
+    const maxSalary = job.salaryRange.split(splittingString)[1];
+    const datePosted = new Date().toISOString();
+
+    const schema = `
+    {
+        "@context": "http://schema.org",
+        "@type": "JobPosting",
+        "estimatedSalary": {
+            "@type": "MonetaryAmount",
+            "currency": "GBP",
+            "value": {
+                "@type": "QuantitativeValue",
+                "minValue": "${minSalary}",
+                "maxValue": "${maxSalary}",
```

Comment:
maxSalary can be undefined if salaryRange doesn't include a ' - ' or ' to '. Maybe we should provide these values depending on if maxSalary exists or not.
It also looks like quantititiveValue is **not** supported by estimatedSalary https://schema.org/estimatedSalary so this might need changing

Reply:
See recent commit for changes